pandas/core/index.py

```diff
@@ -1527,6 +1544,64 @@ def get_indexer(self, target, method=None, limit=None):
 
         return com._ensure_platform_int(indexer)
 
+    def _get_fill_indexer(self, target, method, limit=None):
+        if self.is_monotonic and target.is_monotonic:
```

Comment:
dont' these have to be both monotonic_increasing (or both monotonic_decreasing)? e.g. if one is increasing and one decreasing is that an error?